tests/unit/conftest.py

```diff
@@ -3,19 +3,35 @@
 
 """pytest fixtures for the integration test."""
 
+# pylint: disable=protected-access
+
+import typing
+
 import ops.testing
 import pytest
 
 from charm import FlaskCharm
 
 
 @pytest.fixture(name="harness")
-def harness_fixture():
+def harness_fixture(monkeypatch) -> typing.Generator[ops.testing.Harness, None, None]:
     """Ops testing framework harness fixture."""
-    ops.testing.SIMULATE_CAN_CONNECT = True
     harness = ops.testing.Harness(FlaskCharm)
+    orig_send_signal = ops.testing._TestingPebbleClient.send_signal
 
-    yield harness
+    def patched_send_signal(self, sig, service_names):
+        """Patch a bug in the ops testing framework temporarily."""
+        return orig_send_signal(self, sig, *service_names)
 
+    monkeypatch.setattr(ops.testing._TestingPebbleClient, "send_signal", patched_send_signal)
+    yield harness
     harness.cleanup()
-    ops.testing.SIMULATE_CAN_CONNECT = False
+
+
+@pytest.fixture
+def mock_container_fs(monkeypatch) -> dict[str, str]:
```

Comment:
This is the other reason of using a shared volume and moving the configuration handling to a separate module so the patch isn't needed

Reply:
Please see the discussion about shared volumes.

Reply:
Not the file and exec API.